### DISCUSSION TITLE
Merge Advanced Node Versions and add Collapsed Optionals

### DISCUSSION THREAD

**Discussion by WASasquatch** [comfyanonymous/ComfyUI#309] [Ideas]
This idea would rid the need to have special advance nodes doing the same thing save for a few new parameters. 

For all optional parameters, just collapse them with +/- button or something. This functionality would be greatly appreciated in general for nodes that need optional input, but perhaps you don't want a wall of parameter inputs.

**Comment by GrimblyGorn** [comment]
This would be handy and they could all be at the bottom of the nodes and close like a little drawer maybe. Tucked away unless they were in use. Then perhaps left behind when the drawer of unused optionals was closed. So, possibly re-ordered as well then maybe? :)

I also feel like all the nodes could use a description and little drawer at the bottom for that to display in with perhaps a menu toggle instead for those. But, that's possibly quite separate from this.

**Reply by WASasquatch** [reply]
That's exactly what I envisioned. Maybe for descriptions a hovering div modal that can be defined from the node for each input. It would use the inputs parameters to tell the users it's input ranges for like ints and floats, or options for lists, and then the description. Would save in having to edit the actual node template for descriptions.

**Reply by GrimblyGorn** [reply]
As popup modal tooltips that also saves having to make a toggle for them and having to use such a toggle when you decide you want them. So, ya that's probably a better way to handle those. Does seem like it could still use a toggle in that case though. Tooltips can get old sometimes and become invasive if unwanted so being able to disable them is likely a good option either way I guess.

It does seem more direct to have tips for what you point to on the node, instead of a blurb of all the things about the node.

**Reply by WASasquatch** [reply]
> As popup modal tooltips that also saves having to make a toggle for them and having to use such a toggle when you decide you want them. So, ya that's probably a better way to handle those. Does seem like it could still use a toggle in that case though. Tooltips can get old sometimes and become invasive if unwanted so being able to disable them is likely a good option either way I guess.
> 
> It does seem more direct to have tips for what you point to on the node, instead of a blurb of all the things about the node.

Wouldn't that inherently imply breaking up the whole node into parts to have anything to point at to tip? Otherwise you would be pointing at say a input, which inherently would mean telling the user what the input accepts, it's ranges and a description field to explain in a human readable way. 

Also I find tool tips to be the most unobtrusive way. As they would only activate when hovering X milliseconds or seconds, as apposed to needing to actually click certain things, and toggle it on and off. Additionally, the node template changing, in any way, just for a description would be counter-intuitive to a non-destructive tooltip hover. And at that point, what would be the point of editing the node to have all these toggleable areas when it could be a much more organized tooltip coming from its own functional. 

Adding to the actual code to add tip areas for all elements seems overkill and too much node template noise. And a huge block that expands that would inherently have to encompass all node options would be very intrusive to look at, and wouldn't make sense as a toggle on that right alone, vs summarized tooltips relating to what you are hovering only. 

---

Being able to just add a description field to the input would make adding descriptions super easy:

```python
    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "image": ("IMAGE",),
                "brightness": ("FLOAT", {"default": 0.0, "min": -1.0, "max": 1.0, "step": 0.01, 
					"description": "The brightness value for the image, 0.0 means no adjustment"}),
                "contrast": ("FLOAT", {"default": 1.0, "min": -1.0, "max": 2.0, "step": 0.01,
					"description": "The contrast value for the image, 1.0 means no adjustment"}),
                "saturation": ("FLOAT", {"default": 1.0, "min": 0.0, "max": 5.0, "step": 0.01,
					"description": "The saturation value for the image, 1.0 means no adjustment"}),
                "sharpness": ("FLOAT", {"default": 1.0, "min": -5.0, "max": 5.0, "step": 0.01,
					"description": "The sharpness of the image, 1.0 means no sharpness applied"}),
                "blur": ("INT", {"default": 0, "min": 0, "max": 16, "step": 1,
					"description": "The blurring value of the image, 0 is no blur applied"}),
                "gaussian_blur": ("FLOAT", {"default": 0.0, "min": 0.0, "max": 1024.0, "step": 0.1,
					"description": "The Gaussian blur value of the image, 0.0 is no Gaussian blur applied"}),
                "edge_enhance": ("FLOAT", {"default": 0.0, "min": 0.0, "max": 1.0, "step": 0.01,
					"description": "The edge enhancement value of the image, 0.0 is no edge enhancement applied."}),
            },
        }
```
